src/test/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtilsTest.java

```diff
@@ -36,117 +39,163 @@ public void testIsProperUtilsClass() throws ReflectiveOperationException {
+            getNodeWithParentScope(TokenTypes.LITERAL_STATIC, "static", TokenTypes.CLASS_DEF)));
+    }
+
+    @Test
+    public void testIsInScope() {
+        assertTrue(ScopeUtils.isInScope(getNodeWithParentScope(TokenTypes.LITERAL_PUBLIC,
+                "public", TokenTypes.ANNOTATION_DEF), Scope.PUBLIC));
+        assertFalse(ScopeUtils.isInScope(getNodeWithParentScope(TokenTypes.LITERAL_PROTECTED,
+                "protected", TokenTypes.INTERFACE_DEF), Scope.PRIVATE));
+    }
+
+    @Test
+    public void testSurroundingScopeOfNodeChildOfLiteralNewIsAnoninner() {
+        assertEquals(Scope.ANONINNER, ScopeUtils.getSurroundingScope(
+                getNode(TokenTypes.LITERAL_NEW, TokenTypes.IDENT)));
+    }
+
+    @Test
+    public void testIsInInterfaceBlock() {
+        final DetailAST ast = getNode(TokenTypes.INTERFACE_DEF, TokenTypes.OBJBLOCK,
+                TokenTypes.CLASS_DEF, TokenTypes.MODIFIERS);
+
+        assertTrue(ScopeUtils.isInInterfaceBlock(ast.getParent()));
+        assertFalse(ScopeUtils.isInInterfaceBlock(ast));
+    }
+
+    @Test
+    public void testIsInAnnotationBlock() {
+        final DetailAST ast = getNode(TokenTypes.ANNOTATION_DEF, TokenTypes.OBJBLOCK,
+                TokenTypes.INTERFACE_DEF, TokenTypes.MODIFIERS);
+
+        assertTrue(ScopeUtils.isInAnnotationBlock(ast.getParent()));
+        assertFalse(ScopeUtils.isInAnnotationBlock(ast));
+    }
+
+    @Test
+    public void testisInInterfaceOrAnnotationBlock() {
+        assertTrue(ScopeUtils.isInInterfaceOrAnnotationBlock(
+                getNode(TokenTypes.ANNOTATION_DEF, TokenTypes.OBJBLOCK)));
+        assertTrue(ScopeUtils.isInInterfaceOrAnnotationBlock(
+                getNode(TokenTypes.INTERFACE_DEF, TokenTypes.OBJBLOCK)));
+        assertFalse(ScopeUtils.isInInterfaceOrAnnotationBlock(
+                getNode(TokenTypes.CLASS_DEF, TokenTypes.OBJBLOCK)));
+        assertFalse(ScopeUtils.isInInterfaceOrAnnotationBlock(
+                getNode(TokenTypes.LITERAL_NEW, TokenTypes.IDENT)));
+        assertFalse(ScopeUtils.isInInterfaceOrAnnotationBlock(
+                getNode(TokenTypes.ENUM_DEF, TokenTypes.OBJBLOCK)));
+    }
+
+    private static DetailAST getNode(int... nodeTypes) {
+        DetailAST ast = new DetailAST();
+        ast.setType(nodeTypes[0]);
+        for (int i = 1; i < nodeTypes.length; i++) {
+            final DetailAST astChild = new DetailAST();
+            astChild.setType(nodeTypes[i]);
+            ast.addChild(astChild);
+            ast = astChild;
+        }
+        return ast;
+    }
 
-        Assert.assertFalse(ScopeUtils.isLocalVariableDef(ast));
+    private static DetailAST getNodeWithParentScope(int literal, String scope,
+                                                    int parentTokenType) {
+        final DetailAST ast = getNode(parentTokenType, TokenTypes.MODIFIERS, literal);
+        ast.setText(scope);
+        final DetailAST ast2 = getNode(TokenTypes.OBJBLOCK);
```

Comment:
it is first time when codacy-bot is right and nobody else catch a problem.
@Nimfadora , please fix.